src/contracts/pods/EigenPodManager.sol

```diff
@@ -226,34 +224,37 @@ contract EigenPodManager is
 
     /// @notice Called by proving service to fulfill partial withdrawal proof requests
     function proofServiceCallback(
-        WithdrawalCallbackInfo[] calldata callbackInfo
+        WithdrawalCallbackInfo calldata callbackInfo
     ) external onlyProofService nonReentrant {
         require(proofServiceEnabled, "EigenPodManager.proofServiceCallback: offchain partial withdrawal proofs are not enabled");
-        for(uint256 i = 0; i < callbackInfo.length; i++) {
-            (bytes32 imageID, Journal memory journal, bytes memory journalBytes) = parsePayload(callbackInfo[i].payload);
 
-            require(IRiscZeroVerifier(proofService.verifier).verify(callbackInfo[i].seal, imageID, callbackInfo[i].postStateDigest, sha256(journalBytes)), "EigenPodManager.proofServiceCallback: invalid proof");
+        Journal memory journal = callbackInfo.journal;
+        require(IRiscZeroVerifier(proofService.verifier).verify(callbackInfo.seal, callbackInfo.imageId, callbackInfo.postStateDigest, sha256(abi.encode(journal))), "EigenPodManager.proofServiceCallback: invalid proof");
+
+        require(journal.blockRoot == getBlockRootAtTimestamp(callbackInfo.oracleTimestamp), "EigenPodManager.proofServiceCallback: block root does not match oracleRoot for that timestamp");
+        
+        for (uint256 i = 0; i < journal.podOwners.length; i++) {
 
-            require(journal.blockRoot == getBlockRootAtTimestamp(callbackInfo[i].oracleTimestamp), "EigenPodManager.proofServiceCallback: block root does not match oracleRoot for that timestamp");
-            
             // these checks are verified in the snark, we add them here again as a sanity check
-            require(callbackInfo[i].oracleTimestamp >= journal.endTimestamp, "EigenPodManager.proofServiceCallback: oracle timestamp must be greater than or equal to callback timestamp");
-            require(callbackInfo[i].feeGwei <= journal.maxFeeGwei, "EigenPodManager.proofServiceCallback: fee must be less than or equal to maxFee");
+            require(callbackInfo.oracleTimestamp >= journal.endTimestamps[i], "EigenPodManager.proofServiceCallback: oracle timestamp must be greater than or equal to callback timestamp");
+            require(callbackInfo.feeGwei <= journal.maxFeesGwei[i], "EigenPodManager.proofServiceCallback: fee must be less than or equal to maxFee");
```

Comment:
super confused, how can all of the journals have the same fee  and oracle timestamp? the callback struct needs its internal fields arrayified

Reply:
thats incorrect, fixed